src/Microsoft.ApplicationInsights.AspNet/Extensibility/Implementation/Tracing/TelemetryLoggerScope.cs

```diff
@@ -0,0 +1,37 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="TelemetryLoggerScope.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+namespace Microsoft.ApplicationInsights.AspNet.Extensibility.Implementation.Tracing
+{
+    using System;
+
+    /// <summary>
+    /// Logging scope for <code>TelemetryLogger.BeginScopeImpl</code> calls.
+    /// </summary>
+    public class TelemetryLoggerScope : IDisposable
+    {
```

Comment:
Should you follow basic dispose pattern (https://msdn.microsoft.com/en-us/library/b1yfkh5e(v=vs.110).aspx)